Runtime.php

```diff
@@ -79,20 +79,23 @@ public static function getUri()
      */
     public static function getData($extended = true)
     {
-        switch (static::getMethod()) {
+        $method = static::getMethod();
+
+        switch ($method) {
             case Request::METHOD_GET:
                 return $_GET;
 
             case Request::METHOD_POST:
+            case Request::METHOD_PUT:
+            case Request::METHOD_PATCH:
                 $contentType = static::getHeader('Content-Type');
 
                 switch ($contentType) {
                     case 'application/x-www-form-urlencoded':
-                        return $_POST;
+                        return ${'_'.strtoupper($method)};
```

Comment:
Yeap! 
I will write a first switch for the method and a second switch for the content-type. 
I saw that: https://github.com/nikic/FastRoute/issues/65#issuecomment-122523492
It will explain the next revision.